### ISSUE TITLE
Describe variable fails due to missing definition of "helm-bookmark-map" on develop

### ISSUE THREAD

**Issue by smile13241324** [syl20bnr/spacemacs#9549]
#### Description :octocat:
Since the latest update on develop every call of `SPC h d v` fails due to an undefined variable `helm-bookmark-map`.

#### Reproduction guide :beetle:
- Start Emacs
- Press `SPC h d v` and select a variable of your choice
- The error appears

*Observed behaviour:* :eyes: :broken_heart:
The symbol `helm-bookmark-map` is referenced but not defined which causes emacs to fail at displaying the variable description. A quick smart search showed me the file is referenced from the helm layer's packages.el only.

*Expected behaviour:* :heart: :smile:
Show the description

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.2.1
- Spacemacs: 0.200.9
- Spacemacs branch: develop (rev. 6d73df70)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm nim scheme
      (auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'cycle auto-completion-complete-with-key-sequence nil auto-completion-complete-with-key-sequence-delay 0.0 auto-completion-private-snippets-directory nil auto-completion-enable-snippets-in-popup nil auto-completion-enable-help-tooltip t auto-completion-enable-sort-by-usage t)
      better-defaults
      (clojure :variables clojure-enable-fancify-symbols t)
      gpu graphviz
      (plantuml :variables plantuml-jar-path "~/.plantuml/plantUml.jar" org-plantuml-jar-path "~/.plantuml/plantUml.jar")
      salt ibuffer emacs-lisp pdf-tools cscope gtags
      (org :variables org-want-todo-bindings t org-enable-github-support t org-enable-bootstrap-support t org-enable-reveal-js-support t)
      (shell :variables shell-enable-smart-eshell t shell-default-shell 'ansi-term shell-default-height 60 shell-default-position 'bottom shell-default-full-span nil shell-default-term-shell "/usr/bin/fish")
      syntax-checking version-control semantic themes-megapack
      (c-c++ :variables c-c++-default-mode-for-headers 'c++-mode c-c++-enable-clang-support t c-c++-enable-clang-format-on-save t)
      git github sml pass common-lisp
      (python :variables python-test-runner 'pytest python-enable-yapf-format-on-save t python-sort-imports-on-save t)
      windows-scripts agda jr shell-scripts emoji
      (markdown :variables markdown-live-preview-engine 'vmd)
      bibtex
      (latex :variables latex-enable-auto-fill t latex-enable-magic t latex-enable-folding t)
      lua html
      (javascript :variables javascript-disable-tern-port-files nil)
      react yaml nginx docker csv search-engine restructuredtext selectric octave systemd imenu-list
      (java :variables java-backend 'Meghanada)
      (go :variables go-use-gometalinter t gofmt-command "goimports" go-tab-width 4 go-use-gocheck-for-testing t go-use-test-args "-race -timeout 10s")
      major-modes coq django
      (elm :variables elm-sort-imports-on-save t elm-format-on-save t)
      idris
      (haskell :variables haskell-enable-hindent-style "fundamental" haskell-completion-backend 'dante)
      pandoc sphinx
 parinfer asciidoc ansible cfengine puppet rebox rust xkcd typography vimscript
      (terraform :variables terraform-auto-format-on-save t)
      (ruby :variables ruby-enable-enh-ruby-mode t ruby-version-manager 'rvm)
      ruby-on-rails asm
      (sql :variables sql-capitalize-keywords t)
      perl6 autohotkey
      (elixir :variables flycheck-elixir-credo-strict t)
      faust vagrant erlang protobuf php racket
      (typescript :variables typescript-fmt-tool 'typescript-formatter typescript-fmt-on-save t))
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GCONF GSETTINGS NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (void-variable helm-bookmark-map)
  (define-key helm-bookmark-map (kbd "C-d") (quote helm-bookmark-run-delete))
  simpler-helm-bookmark-keybindings()
  run-hooks(helm-mode-hook helm-mode-on-hook)
  helm-mode(1)
  #[0 "\302\303!\210\300\211\207" [describe-variable this-command helm-mode 1] 2]()
  helm--completing-read-default("Describe variable: " [preview-copy-mml edebug-Trace-fast-mode helm-info-libffi edebug-inside-func helm-source-info-ede helm-info-gnutls sh-test debugger-previous-window semantic-decoration-mode-hook makefile-makepp-font-lock-keywords sfile-p magit-update-vc-modeline at\ point helm-info-automake tramp-compat-temporary-file-directory start-deleted electric-help-undefined edebug-form-begin-marker helm-info-viper edebug-mode-on-hook executable-magicless-file-regexp helm-info-gnutls-guile helm-info-erc edebug-print-trace-before term-dynamic-list-filename-completions edebug-eval-mode-hook helm-show-completion-display-function edebug-mark-
```

**Comment by duianto** [comment]
There's a PR: Fix helm-bookmark initialization #9547

**Comment by smile13241324** [comment]
Thanks for telling me, didn't see that. Lets hope this gets merged fast.

**Comment by bjarkevad** [comment]
Pinning helm to melpa stable seems to work on develop:
```
(defun dotspacemacs/user-init ()
  (push '("melpa-stable" . "stable.melpa.org/packages/") configuration-layer--elpa-archives)
  (push '(helm . "melpa-stable") package-pinned-packages))
```

**Comment by jajpater** [comment]
Thanks for the tip bjarkevad but it doesn't work for me.

**Comment by bjarkevad** [comment]
@jajpater: Did you remove `.emacs.d/elpa/helm-*` and restart spacemacs? This should force a reinstall of helm with the pinned version.

**Comment by jajpater** [comment]
Thanks @bjarkevad That solved it!

**Comment by bmag** [comment]
Should be fixed in https://github.com/syl20bnr/spacemacs/commit/f221d7ed7308d029bb302757161ec51853c461b2, can you please verify?

**Comment by duianto** [comment]
@bmag Confirmed, `C-h v` and saving `SPC f s`, works again.

**Comment by smile13241324** [comment]
@bmag I can also verify that your fix worked, on latest develop I can `SPC h d v` again.

**Comment by jajpater** [comment]
Yep it works. Issue can be closed.

**Comment by bmag** [comment]
If there are any users of master branch who suffer this error: calling `SPC f b` or adding `(require 'helm-bookmark)` to user-config should be a proper work-around.

**Comment by dennismayr** [comment]
Workaround confirmed to work.

Stable setup: Emacs 24.5.1, Spacemacs 0.200.9 (master)

**Comment by duianto** [comment]
This seems to be a big enough issue, that a hotfix might be needed for the master branch.
Here are some of the actions that doesn't work and only show the message:
>simpler-helm-bookmark-keybindings: Symbol’s value as variable is void: helm-bookmark-map

`C-h v test RET`
`C-h f test RET`

Save a new buffer:
open a new empty buffer `SPC b N`
type something: `test`
save: `SPC f s`
choose a location and press `RET`

Switch branch in a magit buffer: `bb`
choose a branch and press `RET`

Some gitter comments also mentioned these:

Syncing layers `SPC f e R`

`M-x` `desktop-save`

Copy a file: `SPC f c`
choose a location and press `RET`

Rename a file: `SPC f R`
choose a new name and press `RET`

`M-x` `magit status`
choose a repository, ex `.emacs.d` and press `RET`

#### System Info :computer:
- OS: windows-nt
- Emacs: 26.0.50
- Spacemacs: 0.200.9
- Spacemacs branch: master (rev. 8e1af145)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey auto-completion colors emacs-lisp git helm markdown org version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES

**Comment by GaryHeath** [comment]
Any news on this? Today I innocently ran "Update Packages" and now Spacemacs is unusable for cider-repl loading (C-c C-l) due to “Symbol’s value as variable is void: helm-bookmark-map”. I tried adding the melpa stable fix and reloading helm but I still get the same error. Latest Spacemacs 0.200.9@25.2.1

**Comment by duianto** [comment]
@GaryHeath Currently, the simplest solution is, as stated in the [earlier comment by bmag](https://github.com/syl20bnr/spacemacs/issues/9549#issuecomment-327788403). 

Add this command: `(require 'helm-bookmark)` to the `dotspacemacs/user-config()` section in your `.spacemacs` file. Then restart Emacs and the `helm-bookmark-map` message will go away.

**Comment by GaryHeath** [comment]
Hi Duianto

Thanks. That worked. Was fixated on the other suggestions and missed 
this one.

Gary

**Comment by bmag** [comment]
> This seems to be a big enough issue, that a hotfix might be needed for the master branch.

Yes, we are aware. syl20bnr is planning to make a hot fix release soon (I don't know exactly when, but hopefully within several days)

**Comment by bmag** [comment]
Reopening for visibility. The issue will be closed in the next release.

**Comment by liuchong** [comment]
fixed in develop? will it take long to be merged into master? 😹

**Comment by liuchong** [comment]
Fixed in develop? Will it take long to merge develop into master? 😹

**Comment by liuchong** [comment]
fixed in develop? will it take long to merge develop to master? 😹

**Comment by liuchong** [comment]
Fixed in develop? Will it take long to merge develop into master? 😹

**Comment by jakemcarthur** [comment]
It seems pretty sad for this not to be in a hotfix.

**Comment by dennismayr** [comment]
I kept getting this error on Emacs 24.5.1 (stable), using Spacemacs (stable) with the pinned, melpa-stable `helm` package.

Adding `(require 'helm-bookmark)` in `.spacemacs` as a workaround fixed it.

**Comment by fosskers** [comment]
This breaks branch switching with `magit` as well.

**Comment by vatrat** [comment]
So I also have this problem, but adding `(require 'helm-bookmark)` doesn't solve the problem. Running `SPC f b` and then closing it works. I've tried this on multiple systems running Fedora 25 and 26. I've tried adding the require line to the top and bottom of user-config, but it doesn't work.

**Comment by fosskers** [comment]
@vatrat are you fully closing and restarting spacemacs after adding `(require 'helm-bookmark)`?

**Comment by vatrat** [comment]
Yes, I've actually had that line in my config for a few weeks. I tried to solve this issue a few weeks ago and didn't get back to it until now

**Comment by mcwienczek** [comment]
I am still having this problem and `(require 'helm-bookmark)` doesn't seem to help. Tried also to `rm -rf .emacs.d/elpa/helm-*` and restarting emacs.

**Comment by vatrat** [comment]
Switching to the development branch did fix this issue, but it seems like overkill. I do like the new features, though.

**Comment by SlightlyCyborg** [comment]
@duianto Ok. Adding the `(require 'helm-bookmark)` worked. It confused me why deleting all of helm and then restarting emacs worked, but then if I restarted again it would fail. It was because spacemacs would download and load the "new" helm package when it was missing, but after helm was installed emacs wouldn't care to load it again on restart.

**Comment by cghtompkins** [comment]
I had the same problem when trying to switch branch with magit. Adding `(require 'helm-bookmark)` worked for me as well.

**Comment by joneshf** [comment]
Chiming in as another person on master, updated packages after a while, and hit this bug.

I'm curious what the status of fixing master is.

**Comment by macalinao** [comment]
Please fix this. It's breaking my workflow.

**Comment by tuxayo** [comment]
@macalinao The problem is known and fixed in the develop branch, no need to add messages which could be replaced by upvoting the issue or some of latest comments. (because they dilute the messages related to how to fix the issue and release the fix)

Someone correct me if I'm wrong, but the best thing to do to recover the functionality + helping to ship it would be to switch on the develop branch and report eventual bugs (or just enjoy the fix). So the next release will come sooner.
Or [funding the project ](https://salt.bountysource.com/teams/spacemacs) so maintainers could spent more time on it.
Or [creating a bounty](https://www.bountysource.com/teams/spacemacs/issues) on this specific issue.

**Comment by tonylotts** [comment]
A newbie just posted on Gitter about this bug.  The fix is trivial for those whom are used to Spacemacs.  However this bug has the potential to make a bad impression on new users.

It has been over a month since it was determined that there should be a hotfix.  What's holding up the hot fix?  Anything I can do to help?

**Comment by dennismayr** [comment]
That's a good point. The bug can be fixed easily...if one has ever had
experience with Emacs.
If I had seen this error 2 months ago, as a newbie, I'd have thought "well,
this doesn't work", and probably switched to Atom or Sublime.

**Comment by smile13241324** [comment]
As I see that many of you are getting pretty frustrated with the state on master please consider:
• We are all doing this work in our freetime
• ‎We all have jobs
• ‎We all have families

To be honest currently we have more issues then we have hands to fix. Therefore everybody focus on develop rather than on the old master.

Speaking of hands ... you know spacemacs is community driven, that means we expect and need our users to get involved in the project.

So if you like spacemacs please consider
• If your the programming type fetch one of the random issues from the issue tracker like your generic org export issue or some layer not working properly ticket.
• ‎If you are unsure pick one of the various documentation issues from #9476. This is also a pretty good way to get more familiar with your editor. 
• ‎If you are unsure of what is to be done for an issue you can still ask one of the team normally we are pretty fast in responding to sincere questions.

Consider the more stuff we contributors fix the more time the colaborators have for making hotfixes and working on the core. And this is what you guys are all waiting for. So please get involved.

**Comment by tonylotts** [comment]
Good points @smile13241324.  

We're all developers here, and I assume we all have a good understanding of what 'hotfix' means.   This bug is labelled as fixed on develop.  What this tells me, is that the fix is waiting on someone to merge or cherry-pick the specific parts needed for a release.

As you said this is a community driven project.  The community has fixed the bug; and we are waiting on someone to do the actual release.  If there is anything holding this back, please list it here so we (collectively) can contribute toward this fix.

**Comment by smile13241324** [comment]
@syl20bnr Can you tell when you will be able to provide a hotfix for this on master? This issue seems to become pretty urgent, judging from the amount of people commenting here. 

Is there anything blocking you where we could help?

**Comment by tuxayo** [comment]
Has anyone tried to backport this fix to master an check that it doesn't break anything? That's one step that the maintainer.s wouldn't have to do.

**Comment by SlightlyCyborg** [comment]
I can prepare the hotfix and do a backport, since I am installing spacemacs on a new machine today. What is the commit# in develop?

**Comment by vbauerster** [comment]
I'm a neovim user trying out spacemacs at the moment. The issue affects go layer: most of go-guru commands do not work out of the box. Also customize-variable do not popup preference window.
Hope you fix this soon, because it is very discouraging for newcommers.

**Comment by tuxayo** [comment]
@SlightlyCyborg 
> What is the fix's commit# in develop?

It seems to be f221d7ed7308d029bb302757161ec51853c461b2

**Comment by syl20bnr** [comment]
Going to release a 200.10 with the hotfix.

**Comment by antonpaisov** [comment]
@syl20bnr do you mean today or in general? :)

**Comment by syl20bnr** [comment]
Planned tomorrow sorry, yesterday I worked on the signing of Spacelpa tarballs.

Hum ^_^

**Comment by syl20bnr** [comment]
Yesterday you said tomorrow so just DO IT!

Fixed in [0.200.10](https://github.com/syl20bnr/spacemacs/releases/tag/v0.200.10) 🎉

**Comment by fosskers** [comment]
Yup, seems to work!

**Comment by antonpaisov** [comment]
time to close this issue :)

**Comment by TangMonk** [comment]
after I update to 0.200.10, another error throw: 

```Error (use-package): winum :config: Symbol’s value as variable is void: winum-assign-functions```

**Comment by syl20bnr** [comment]
Fixed in both branches.
